### PR TITLE
AgentStore Host Service

### PR DESCRIPTION
**Pull request by azdagron**
This PR contains the AgentStore host service implementation, which is used by server-side node attestors to query for agent attestation status in #922. It is being pulled out in it's own PR to help make #922 easier to review.

### CODE REVIEW

File: pkg/server/hostservices/agentstore/agentstore.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"sync"
+
+	"github.com/spiffe/spire/proto/spire/server/datastore"
+	"github.com/spiffe/spire/proto/spire/server/hostservices"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Deps struct {
+	// DataStore is used to retrieve the lastest bundle. It MUST be set.
```
Comment (evan2645) [inline pkg/server/hostservices/agentstore/agentstore.go]: I think this comment is stale?

**Comment by ZymoticB** [comment]
Just coming back from vacation. If this is a blocker feel free to merge, I'll try to take a look in the next 48 hours.

File: proto/spire/server/hostservices/README_pb.md [hunk 2]
```diff
@@ -3,63 +3,66 @@
 
 ## Table of Contents
 
-- [identityprovider.proto](#identityprovider.proto)
-    - [FetchX509IdentityRequest](#spire.server.hostservices.FetchX509IdentityRequest)
-    - [FetchX509IdentityResponse](#spire.server.hostservices.FetchX509IdentityResponse)
-    - [X509Identity](#spire.server.hostservices.X509Identity)
+- [agentstore.proto](#agentstore.proto)
```
Comment (ZymoticB) [inline proto/spire/server/hostservices/README_pb.md]: This seems to be replacing the identity provider documentation, is that intentional?

Comment (azdagron) [reply proto/spire/server/hostservices/README_pb.md]: Yeah that's weird...

Comment (azdagron) [reply proto/spire/server/hostservices/README_pb.md]: Oh yep. The goo in build.sh in charge of generating the README does not handle more than one .proto in a directory and ends up overwriting. I'll fix that real quick.

**Pull request merged**